### PR TITLE
Standardize colon spacing in docstrings

### PR DESCRIPTION
**Pull request by chrishalcrow**
Standardize colon spacings in docstrings.

All public functions now use the numpydoc colon spacing (https://numpydoc.readthedocs.io/en/latest/format.html#parameters). Meaning that they look like
```
recording : RecordingExtractor
    The recording extractor object to be saved in .dat format
```
rather than
```
recording: RecordingExtractor
    The recording extractor object to be saved in .dat format
```

This means we can use the `numpydoc_validation_checks = {"PR01"}` sphinx parameter to easily check for conflicts between function/class arguments and their documented names.

The format is also consistent with the standard numpydoc autodoc generator (https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring).

### CODE REVIEW

**Review by zm711 — COMMENTED** [review]
Fix the default `:`

File: src/spikeinterface/comparison/paircomparisons.py [hunk 1]
```diff
@@ -664,24 +664,24 @@ def count_units_categories(
...
+GT num_units : {num_gt}
+TESTED num_units : {num_tested}
+num_well_detected : {num_well_detected}
+num_redundant : {num_redundant}
+num_overmerged : {num_overmerged}
 """
 
 _template_summary_part2 = """num_false_positive_units {num_false_positive_units}
-num_bad: {num_bad}
+num_bad : {num_bad}
```
Comment (alejoe91) [inline src/spikeinterface/comparison/paircomparisons.py:684]: @chrishalcrow these are for printing the performance, so I think we can revert

**Review by alejoe91 — CHANGES_REQUESTED** [review]
Thanks @chrishalcrow 

Just a little comment!

File: src/spikeinterface/core/sparsity.py [hunk 2]
```diff
@@ -11,29 +11,29 @@
 
 
 _sparsity_doc = """
-    method: str
-        * "best_channels": N best channels with the largest amplitude. Use the "num_channels" argument to specify the
+    method : str
```
Comment (zm711) [inline src/spikeinterface/core/sparsity.py:14]: Since this is an injected docstring does the numpydoc validation still work the same?

**Review by alejoe91 — APPROVED** [review]

**Review by zm711 — APPROVED** [review]
This looks great to me. Chris could you remember to also change our docstring contributing guide to account for this?

**Comment by alejoe91** [comment]
Thanks @chrishalcrow ! Let's merge :)

**Comment by samuelgarcia** [comment]
Thanks a lot Chris for this super tedious but usefull fight.

**Pull request merged**